src/main/java/uk/gov/hmcts/ccd/domain/service/supplementarydata/DefaultSupplementaryDataOperation.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.ccd.domain.service.supplementarydata;
+
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.casedetails.supplementarydata.Operation;
+import uk.gov.hmcts.ccd.data.casedetails.supplementarydata.SupplementaryDataRepository;
+import uk.gov.hmcts.ccd.domain.model.std.SupplementaryData;
+import uk.gov.hmcts.ccd.domain.model.std.SupplementaryDataUpdateRequest;
+
+@Service
+@Qualifier("default")
+public class DefaultSupplementaryDataOperation implements SupplementaryDataOperation {
+
+    private final SupplementaryDataRepository supplementaryDataRepository;
+
+    @Autowired
+    public DefaultSupplementaryDataOperation(final @Qualifier("default") SupplementaryDataRepository supplementaryDataRepository) {
+        this.supplementaryDataRepository = supplementaryDataRepository;
+    }
+
+    @Override
+    public SupplementaryData updateSupplementaryData(String caseReference, SupplementaryDataUpdateRequest supplementaryData) {
+        incrementData(caseReference, supplementaryData);
+        setData(caseReference, supplementaryData);
+        return this.supplementaryDataRepository.findSupplementaryData(caseReference);
```

Comment:
are we returning the whole supplementary data here? we should return only the updated properties

Reply:
Yes, we are returning whole supplementary data.. just had a chat with Dil on this.. he said whole supplementary data